drivers/storage/portworx/component/telemetry.go

```diff
@@ -0,0 +1,170 @@
+package component
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	pxutil "github.com/libopenstorage/operator/drivers/storage/portworx/util"
+	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// TelemetryComponentName name of the telemetry component
+	TelemetryComponentName = "Portworx Telemetry"
+	// TelemetryConfigMapName is the name of the config map that stores the telemetry configuration for CCM
+	TelemetryConfigMapName = "px-telemetry-config"
+	// TelemetryPropertiesFilename is the name of the CCM properties file
+	TelemetryPropertiesFilename = "ccm.properties"
+	// TelemetryArcusLocationFilename is name of the file storing the location of arcus endpoint to use
+	TelemetryArcusLocationFilename = "location"
+)
+
+type telemetry struct {
+	isCreated bool
```

Comment:
We don't need this for our use case. 
It is added for components involving Deployments as taking a diff of deployments is difficult, so we use this flag.

Reply:
removed